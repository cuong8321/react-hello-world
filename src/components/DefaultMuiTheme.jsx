import React from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import {darkBaseTheme as defaultTheme} from 'material-ui/styles/baseThemes/darkBaseTheme'
import getMuiTheme from 'material-ui/styles/getMuiTheme'

export default function DefaultMuiTheme (props) {
  return <MuiThemeProvider muiTheme={getMuiTheme(defaultTheme)}>
    {props.children}
  </MuiThemeProvider>
}
