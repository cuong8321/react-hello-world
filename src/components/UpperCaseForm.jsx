import React from 'react'
import TextField from 'material-ui/TextField'
import FlatButton from 'material-ui/FlatButton'
import Paper from 'material-ui/Paper'

export default class UpperCaseForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      text: ''
    }
  }

  render () {
    const onChange = (event, text) => this.setState({text})
    const clear = () => this.setState({text: ''})

    return <Paper zDepth={1}><div>
      <div>
        <TextField id='input' hintText='Your Text Here...' onChange={onChange} value={this.state.text} />
        <FlatButton secondary label='Clear' onClick={clear} />
        <Paper zDepth={0}>
          <label htmlFor='output-upper'>Upper Case: </label>
          <output id='output-upper'>{this.state.text.toUpperCase() || <i>(Empty)</i>}</output>
        </Paper>
        <Paper zDepth={0}>
          <label htmlFor='output-lower'>Lower Case: </label>
          <output id='output-lower'>{this.state.text.toLowerCase() || <i>(Empty)</i>}</output>
        </Paper>
      </div>
      <div>
        <Paper zDepth={0} />
      </div>
    </div></Paper>
  }
}
