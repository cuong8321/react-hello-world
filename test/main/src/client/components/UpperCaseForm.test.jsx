import React from 'react'
import renderer from 'react-test-renderer'
import UpperCaseForm from '../../../../../src/client/components/UpperCaseForm.jsx'
import DefaultMuiTheme from '../../../../../src/client/components/DefaultMuiTheme.jsx'

test('Snapshot: components/UpperCaseForm.jsx', () => {
  const tree = renderer.create(<DefaultMuiTheme>
    <UpperCaseForm />
  </DefaultMuiTheme>).toJSON()
  expect(tree).toMatchSnapshot()
})
