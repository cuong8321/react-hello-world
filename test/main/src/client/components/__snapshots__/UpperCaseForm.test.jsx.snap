// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/UpperCaseForm.jsx 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <div>
    <div>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 256,
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Your Text Here...
        </div>
        <input
          disabled={false}
          id="undefined-YourTextHere-undefined-39827"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#ff4081",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Clear
          </span>
        </div>
      </button>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": undefined,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <label
          htmlFor="output"
        >
          Upper Case: 
        </label>
        <output
          id="output"
        >
          <i>
            (Empty)
          </i>
        </output>
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": undefined,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      />
    </div>
  </div>
</div>
`;
